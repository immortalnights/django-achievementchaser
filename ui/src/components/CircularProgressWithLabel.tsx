import { Star } from "@mui/icons-material"
import {
    Box,
    CircularProgress,
    CircularProgressProps,
    Typography,
} from "@mui/material"

// from https://mui.com/material-ui/react-progress/#circular-with-label
const CircularProgressWithLabel = (
    props: CircularProgressProps & { value: number }
) => {
    return (
        <Box margin="auto">
            <Box
                sx={{
                    position: "relative",
                    display: "inline-flex",
                }}
            >
                <CircularProgress
                    sx={{ position: "absolute", color: "lightgray" }}
                    variant="determinate"
                    value={100}
                />
                <CircularProgress variant="determinate" {...props} />
                <Box
                    sx={{
                        top: 0,
                        left: 0,
                        bottom: 0,
                        right: 0,
                        position: "absolute",
                        display: "flex",
                        alignItems: "center",
                        justifyContent: "center",
                    }}
                >
                    {props.value === 100 ? (
                        <Star htmlColor="gold" />
                    ) : (
                        <Typography
                            variant="caption"
                            component="div"
                            color="text.secondary"
                        >{`${Math.floor(props.value)}%`}</Typography>
                    )}
                </Box>
            </Box>
        </Box>
    )
}

export default CircularProgressWithLabel
