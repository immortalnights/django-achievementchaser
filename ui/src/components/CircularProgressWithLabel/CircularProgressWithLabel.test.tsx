import renderer from "react-test-renderer"
import { expect, test } from "vitest"
import CircularProgressWithLabel from "."

test("snapshot", () => {
    // eslint-disable-next-line
    const component = renderer.create(
        <CircularProgressWithLabel value={50} color="primary" />
    )
    expect(component.toJSON()).toMatchSnapshot()
})
