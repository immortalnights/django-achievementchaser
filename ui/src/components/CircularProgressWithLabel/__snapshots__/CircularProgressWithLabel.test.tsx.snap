// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1wd9tnf{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;position:relative;width:40px;height:40px;}",
      }
    }
    data-emotion="css 1wd9tnf"
  />,
  <div
    className="MuiBox-root css-1wd9tnf"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1ake8e4-MuiCircularProgress-root{display:inline-block;-webkit-transition:-webkit-transform 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:transform 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;color:#1976d2;position:absolute;color:lightgray;}",
        }
      }
      data-emotion="css 1ake8e4-MuiCircularProgress-root"
    />
    <span
      aria-valuenow={100}
      className="MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorPrimary css-1ake8e4-MuiCircularProgress-root"
      role="progressbar"
      style={
        {
          "height": 40,
          "transform": "rotate(-90deg)",
          "width": 40,
        }
      }
    >
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-1idz92c-MuiCircularProgress-svg{display:block;}",
          }
        }
        data-emotion="css 1idz92c-MuiCircularProgress-svg"
      />
      <svg
        className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-oxts8u-MuiCircularProgress-circle{stroke:currentColor;-webkit-transition:stroke-dashoffset 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:stroke-dashoffset 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;}",
            }
          }
          data-emotion="css oxts8u-MuiCircularProgress-circle"
        />
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate css-oxts8u-MuiCircularProgress-circle"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={
            {
              "strokeDasharray": "126.920",
              "strokeDashoffset": "0.000px",
            }
          }
        />
      </svg>
    </span>
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1rn30mb-MuiCircularProgress-root{display:inline-block;-webkit-transition:-webkit-transform 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:transform 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;color:#1976d2;}",
        }
      }
      data-emotion="css 1rn30mb-MuiCircularProgress-root"
    />
    <span
      aria-valuenow={50}
      className="MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorPrimary css-1rn30mb-MuiCircularProgress-root"
      role="progressbar"
      style={
        {
          "height": 40,
          "transform": "rotate(-90deg)",
          "width": 40,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate css-oxts8u-MuiCircularProgress-circle"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={
            {
              "strokeDasharray": "126.920",
              "strokeDashoffset": "63.460px",
            }
          }
        />
      </svg>
    </span>
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1o8dslu{top:0;left:0;bottom:0;right:0;position:absolute;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;}",
        }
      }
      data-emotion="css 1o8dslu"
    />
    <div
      className="MuiBox-root css-1o8dslu"
    >
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-ol7wxs-MuiTypography-root{margin:0;font-family:\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif;font-weight:400;font-size:0.75rem;line-height:1.66;letter-spacing:0.03333em;color:rgba(0, 0, 0, 0.6);}",
          }
        }
        data-emotion="css ol7wxs-MuiTypography-root"
      />
      <div
        className="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
      >
        50%
      </div>
    </div>
  </div>,
]
`;
